src/cloud-api-adaptor/cmd/agent-protocol-forwarder/main.go

```diff
@@ -63,23 +68,30 @@ func (cfg *Config) Setup() (cmd.Starter, error) {
 		flags.StringVar(&tlsConfig.KeyFile, "cert-key", "", "cert key")
 		flags.BoolVar(&tlsConfig.SkipVerify, "tls-skip-verify", false, "Skip TLS certificate verification - use it only for testing")
 		flags.BoolVar(&disableTLS, "disable-tls", false, "Disable TLS encryption - use it only for testing")
+		flags.BoolVar(&secureComms, "secure-comms", false, "Use SSH to secure communication between cluster and peer pods")
+		flags.StringVar(&secureCommsInbounds, "secure-comms-inbounds", "", "Inbound tags for secure communication tunnels")
+		flags.StringVar(&secureCommsOutbounds, "secure-comms-outbounds", "", "Outbound tags for secure communication tunnels")
 	})
 
-	if !disableTLS {
-		cfg.tlsConfig = &tlsConfig
-	}
-
 	cmd.ShowVersion(programName)
 
 	if showVersion {
 		cmd.Exit(0)
 	}
 
-	for path, obj := range map[string]interface{}{
-		cfg.configPath: &cfg.daemonConfig,
-	} {
-		if err := load(path, obj); err != nil {
-			return nil, err
+	if err := load(cfg.configPath, &cfg.daemonConfig); err != nil {
+		return nil, err
+	}
```

Comment:
Nice.. not sure why we had the loop in the first place :-( ..

Reply:
I was bewildered about this code at first... :)